### PR TITLE
feat: create end to end test for squads v7 initiative 4

### PR DESCRIPTION
**Pull request by TomDowling**
## Changes
Adding improved tests for both the ShareModal and ShareBar following the changes in v7, these should cover most of the use cases involved with these components.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1471 #done

### CODE REVIEW

File: packages/shared/src/components/modals/ShareModal.spec.tsx [hunk 1]
```diff
@@ -23,76 +38,141 @@ Object.assign(navigator, {
...
 
-it('should render the article card', async () => {
-  renderComponent();
-  expect(screen.getByAltText(`${defaultPost.title}`)).toBeInTheDocument();
-  expect(
-    screen.getByAltText(`Avatar of ${defaultPost.source.handle}`),
-  ).toBeInTheDocument();
-  expect(screen.getByText(defaultPost.title)).toBeInTheDocument();
-});
+describe('ShareModal Test Suite:', () => {
```
Comment (rebelchris) [inline packages/shared/src/components/modals/ShareModal.spec.tsx:80]: Just a future note, this is slightly redundant as the file will act as a suite already.
Normally we only use describe when it's multi purpose tests inside a file.

But you can leave it for now.

File: packages/shared/src/components/modals/ShareModal.spec.tsx [hunk 2]
```diff
@@ -23,76 +38,141 @@ Object.assign(navigator, {
...
+    renderComponent(true, false);
+    const btn = await screen.findByTestId('social-share-New Squad');
 
-it('should render the copy link section for comments', async () => {
-  renderComponent(defaultComment);
-  const btn = await screen.findByTestId('social-share-Copy link');
+    expect(btn).toBeInTheDocument();
+    btn.click();
+    await waitFor(() => {
+      // expect(screen.getByPlaceholderText('Name your squad')).toBeInTheDocument();
```
Comment (rebelchris) [inline packages/shared/src/components/modals/ShareModal.spec.tsx:103]: This should either be active or removed.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Please make sure the tests are good to go one failing I see (could be flaky)

File: packages/shared/src/components/modals/ShareModal.spec.tsx (hunk 1 above)
Comment (TomDowling) [reply packages/shared/src/components/modals/ShareModal.spec.tsx:81]: Thank you, I'll follow this going forward. Iv'e always used the describe to group the tests even if we are only testing one thing in the file

**Review by rebelchris — APPROVED** [review]

**Pull request merged**